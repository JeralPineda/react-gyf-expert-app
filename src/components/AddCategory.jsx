import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        //
        type="text"
        placeholder="CategorÃ­a"
        value={inputValue}
        onChange={onChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
