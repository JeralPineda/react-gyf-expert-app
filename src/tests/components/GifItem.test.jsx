import { render } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Pruebas en <GifItem />', () => {
  const title = 'One Punch Man';
  const url = 'https://giphy.com/gifs/yo3TC0yeHd53G';

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(
      <GifItem
        //
        title={title}
        url={url}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
