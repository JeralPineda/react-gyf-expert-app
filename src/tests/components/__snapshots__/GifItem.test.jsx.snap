// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man"
      src="https://giphy.com/gifs/yo3TC0yeHd53G"
    />
    <p>
      One Punch Man
    </p>
  </div>
</div>
`;
